src/charm.py

```diff
@@ -1599,6 +1628,8 @@ def _can_run_on_update_status(self) -> bool:
 
         if self.has_raft_keys():
             logger.debug("Early exit on_update_status: Raft recovery in progress")
+            if self.unit.is_leader():
+                self._raft_reinitialisation()
```

Comment:
Should we do this in an automated way or only leave this operation to the action?

Reply:
No, only with the action. Removed.